### PR TITLE
Add proximity sensor example

### PR DESCRIPTION
**Pull request by GregAC**
This requires an updated cheriot-rtos that has https://github.com/lowRISC/cheriot-rtos/pull/3/files and https://github.com/lowRISC/cheriot-rtos/pull/2 merged.

So DO NOT MERGE yet. Once those changes are in we can update the submodule as part of this PR.

### CODE REVIEW

File: cheriot-rtos
[the remark this answers is not in the document]
Comment (GregAC) [reply cheriot-rtos:1]: This is now the correct sub-module bump

File: compartments/proximity_sensor_example.cc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+static void setup_proximity_sensor(Mmio<OpenTitanI2c> i2c, const uint8_t addr) {
+  uint8_t buf[2];
+
+  buf[0] = APDS9960_ID;
+  i2c->write(APDS9960_I2C_ADDRESS, buf, 1, false);
+  if (!i2c->read(APDS9960_I2C_ADDRESS, buf, 1))
+  {
+    Debug::log("Failed to read proximity sensor ID");
+  }
```
Comment (HU90m) [inline compartments/proximity_sensor_example.cc:34]:
```suggestion
  i2c->write(APDS9960_I2C_ADDRESS, buf, 1, false);
  bool success = i2c->read(APDS9960_I2C_ADDRESS, buf, 1);
  Debug::Assert(!success, "Failed to read proximity sensor ID");
```

File: compartments/proximity_sensor_example.cc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	Debug::log("Proximity sensor ID {}", buf[0]);
+
+  if (buf[0] != APDS9960_ID_EXP) {
+    Debug::log("Proximity sensor ID was not expected value of {}", APDS9960_ID_EXP);
```
Comment (HU90m) [inline compartments/proximity_sensor_example.cc:39]: Use `Debug::Assert` here as well?

**Review by HU90m — APPROVED** [review]

File: compartments/proximity_sensor_example.cc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+#include <thread.h>
+
+const uint8_t APDS9960_ENABLE = 0x80;
+const uint8_t APDS9960_ID     = 0x92;
+const uint8_t APDS9960_PPC    = 0x8E;
+const uint8_t APDS9960_CR1    = 0x8F;
+const uint8_t APDS9960_PDATA  = 0x9C;
+
+const uint8_t APDS9960_ID_EXP = 0xAB;
+const uint8_t APDS9960_I2C_ADDRESS = 0x39;
```
[the remark this answers is not in the document]
Comment (GregAC) [reply compartments/proximity_sensor_example.cc:18]: Decided not to do this as I then had to typecast all of the I2C stuff to write it into the buffer and couldn't be bothered :P

**Comment by GregAC** [comment]
Deleting and recreating to hopefully unwedge whatever bit of github has fallen over

**Pull request closed without merging**